integration_tests/tests/invoice-athena-tests.ts

```diff
@@ -3,79 +3,131 @@ import {
   queryObject,
 } from "../helpers/athenaHelper";
 import {
-  putObjectToS3,
   checkIfS3ObjectExists,
-  s3GetObjectsToArray,
   S3Object,
+  s3GetObjectsToJson,
+  deleteDirectoryRecursiveInS3,
+  getS3ItemsList,
+  BillingStandardised,
 } from "../helpers/s3Helper";
 import { resourcePrefix } from "../helpers/envHelper";
-import path from "path";
-import fs from "fs";
+
+import {
+  makeMockInvoicePDF,
+  randomInvoice,
+  writeInvoiceToS3,
+} from "../helpers/mock-data/invoice";
+import { waitForTrue } from "../helpers/commonHelpers";
 
 const prefix = resourcePrefix();
 
 describe("\nExecute athena query to retrieve invoice data and validate that it matches invoice files in storage s3 bucket\n", () => {
   const folderPrefix = "btm_billing_standardised";
-  const testObject: S3Object = {
-    bucket: `${prefix}-storage`,
-    key: `${folderPrefix}/receipt.txt`,
-  };
+  const storageBucket = `${prefix}-storage`;
+
   const databaseName = `${prefix}-invoices`;
+  const invoice = randomInvoice();
 
   beforeAll(async () => {
-    // uploading file to s3 will be removed once BTM-276 implemented
-    const file = "../payloads/receipt.txt";
-    const filePath = path.join(__dirname, file);
-    const fileStream = fs.createReadStream(filePath);
-    await putObjectToS3(testObject, fileStream);
-    const checkFileExists = await checkIfS3ObjectExists(testObject);
-    expect(checkFileExists).toBeTruthy();
+    await deleteDirectoryRecursiveInS3(storageBucket, folderPrefix);
+    const { bucketName, path } = await makeMockInvoicePDF(writeInvoiceToS3)(
+      invoice
+    );
+    const checkRawPdfFileExists = await checkIfS3ObjectExists({
+      bucket: bucketName,
+      key: path,
+    });
+    expect(checkRawPdfFileExists).toBeTruthy();
+    console.log("file exists in raw invoice pdf bucket");
+
+    const checkSizeOfStandardisedFolder = async (): Promise<boolean> => {
+      const result = await getS3ItemsList(storageBucket, folderPrefix);
+      if (result.Contents === undefined) {
+        return false;
+      }
+      return result.Contents.length > 0;
+    };
+
+    const checkIfStandardisedFileExists = await waitForTrue(
+      checkSizeOfStandardisedFolder,
+      1000,
+      20000
+    );
+    expect(checkIfStandardisedFileExists).toBeTruthy();
+    console.log("file exists in standardised folder");
```

Comment:
yes to get the correct results in billing curated view i am deleting the directory otherwise the view will not have expected result for the client which we are testing. Need to implement clean up step after each test so that we do not need to delete the directory there is a ticket in the backlog to put the clean up after each test.For now just reused the existing delete directory. Also, happy to change the approach if there is any to get the correct billing curated view for the client which we are testing